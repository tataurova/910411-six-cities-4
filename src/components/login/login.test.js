import React, {createRef} from "react";
import renderer from "react-test-renderer";
import Login from "./login.jsx";
import {BrowserRouter} from "react-router-dom";

describe(`<Login />`, () => {
  it(`Should Login render correctly`, () => {
    const state = {
      loginValid: null,
      passwordValid: null,
    };
    const loginRef = createRef();
    const passwordRef = createRef();
    const tree = renderer
      .create(
          <BrowserRouter>
            <Login
              state = {state}
              onSubmit = {() => {}}
              onChange = {() => {}}
              loginRef = {loginRef}
              passwordRef = {passwordRef}
            />
          </BrowserRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should Login render correctly with valid form`, () => {
    const state = {
      loginValid: true,
      passwordValid: true,
    };
    const loginRef = createRef();
    const passwordRef = createRef();
    const tree = renderer
      .create(
          <BrowserRouter>
            <Login
              state = {state}
              onSubmit = {() => {}}
              onChange = {() => {}}
              loginRef = {loginRef}
              passwordRef = {passwordRef}
            />
          </BrowserRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
