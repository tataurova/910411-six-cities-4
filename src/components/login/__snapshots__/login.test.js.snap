// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Should Login render correctly 1`] = `
<div
  className="page page--gray page--login"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--login"
  >
    <div
      className="page__login-container container"
    >
      <section
        className="login"
      >
        <h1
          className="login__title"
        >
          Sign in
        </h1>
        <form
          action="#"
          className="login__form form"
          method="post"
          onSubmit={null}
        >
          <div
            className="login__input-wrapper form__input-wrapper"
          >
            <label
              className="visually-hidden"
            >
              E-mail
            </label>
            <input
              className="login__input form__input "
              minLength={6}
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              type="email"
            />
          </div>
          <div
            className="login__input-wrapper form__input-wrapper"
          >
            <label
              className="visually-hidden"
            >
              Password
            </label>
            <input
              className="login__input form__input "
              minLength={1}
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </div>
          <a
            href="/"
            onClick={[Function]}
          >
            <button
              className="login__submit form__submit button"
              type="submit"
            >
              Sign in
            </button>
          </a>
        </form>
      </section>
      <section
        className="locations locations--login locations--current"
      >
        <div
          className="locations__item"
        >
          <a
            className="locations__item-link"
            href="#"
          >
            <span>
              Amsterdam
            </span>
          </a>
        </div>
      </section>
    </div>
  </main>
</div>
`;
